pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -106,54 +116,86 @@ func NewInstanceSettings() datasource.InstanceFactoryFunc {
+	if err != nil {
+		return datasourceService{}, err
+	}
+	s.Services[routeName] = datasourceService{
+		URL:        dsInfo.Routes[routeName].URL,
+		HTTPClient: client,
+	}
+
+	return s.Services[routeName], nil
 }
 
-func newExecutor(im instancemgmt.InstanceManager, cfg *setting.Cfg, executors map[string]azDatasourceExecutor) *datasource.QueryTypeMux {
+func (s *Service) getDSAssetsFromPluginReq(ctx context.Context, req *backend.QueryDataRequest, dsName string) (datasourceInfo, datasourceService, error) {
+	dsInfo, err := s.getDataSourceFromPluginReq(req)
+	if err != nil {
+		return datasourceInfo{}, datasourceService{}, err
+	}
+	service, err := s.getDatasourceService(ctx, dsInfo, dsName)
+	if err != nil {
+		return datasourceInfo{}, datasourceService{}, err
+	}
+	return dsInfo, service, nil
+}
+
+func (s *Service) newMux(executors map[string]azDatasourceExecutor) *datasource.QueryTypeMux {
 	mux := datasource.NewQueryTypeMux()
 	for dsType := range executors {
 		// Make a copy of the string to keep the reference after the iterator
 		dst := dsType
 		mux.HandleFunc(dsType, func(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-			i, err := im.Get(req.PluginContext)
+			executor := executors[dst]
+			dsInfo, service, err := s.getDSAssetsFromPluginReq(ctx, req, dst)
 			if err != nil {
 				return nil, err
 			}
-			dsInfo := i.(datasourceInfo)
-			dsInfo.OrgID = req.PluginContext.OrgID
-			ds := executors[dst]
-			if _, ok := dsInfo.Services[dst]; !ok {
-				// Create an HTTP Client if it has not been created before
-				route := dsInfo.Routes[dst]
-				client, err := newHTTPClient(ctx, route, dsInfo, cfg)
-				if err != nil {
-					return nil, err
-				}
-				dsInfo.Services[dst] = datasourceService{
-					URL:        dsInfo.Routes[dst].URL,
-					HTTPClient: client,
-				}
-			}
-			return ds.executeTimeSeriesQuery(ctx, req.Queries, dsInfo)
+			return executor.executeTimeSeriesQuery(ctx, req.Queries, dsInfo, service.HTTPClient, service.URL)
 		})
 	}
 	return mux
 }
 
 func (s *Service) Init() error {
-	im := datasource.NewInstanceManager(NewInstanceSettings())
+	s.im = datasource.NewInstanceManager(NewInstanceSettings())
 	executors := map[string]azDatasourceExecutor{
 		azureMonitor:       &AzureMonitorDatasource{},
 		appInsights:        &ApplicationInsightsDatasource{},
 		azureLogAnalytics:  &AzureLogAnalyticsDatasource{},
 		insightsAnalytics:  &InsightsAnalyticsDatasource{},
 		azureResourceGraph: &AzureResourceGraphDatasource{},
 	}
+	mux := s.newMux(executors)
+	resourceMux := http.NewServeMux()
+	s.registerRoutes(resourceMux)
 	factory := coreplugin.New(backend.ServeOpts{
-		QueryDataHandler: newExecutor(im, s.Cfg, executors),
+		QueryDataHandler:    mux,
+		CallResourceHandler: httpadapter.New(resourceMux),
 	})
 
-	if err := s.BackendPluginManager.Register(dsName, factory); err != nil {
+	if err := s.BackendPluginManager.RegisterAndStart(context.Background(), dsName, factory); err != nil {
```

Comment:
I believe that `Start` does nothing for core plugins: https://github.com/grafana/grafana/blob/main/pkg/plugins/backendplugin/coreplugin/core_plugin.go#L43

I was just testing this. Since it seems that it's not necessary I will revert this change.